jboss-eap-cd-openshift-launch/added/launch/messaging.sh

```diff
@@ -115,140 +133,169 @@ function configure_mq() {
 }
 
-# $1 - factory name - activemq-ra-remote
-# $2 - username
-# $3 - password
-# $4 - default connection factory - java:jboss/DefaultJMSConnectionFactory
+# $1 - name - messaging-remote-throughput
+# <!-- ##AMQ_REMOTE_CONNECTOR## -->
+generate_remote_artemis_remote_connector() {
+    declare name="${1}"
+    declare socket_binding_name="${2}"
+    echo "<remote-connector name=\"${name}\" socket-binding=\"${socket_binding_name}\"/>" | sed -e ':a;N;$!ba;s|\n|\\n|g'
+}
+
 # <!-- ##AMQ_POOLED_CONNECTION_FACTORY## -->
-function generate_remote_artemis_connection_factory() {
-    echo "<pooled-connection-factory user=\"${2}\" password=\"${3}\" name=\"${1}\" entries=\"java:/JmsXA java:/RemoteJmsXA java:jboss/RemoteJmsXA ${4}\" connectors=\"netty-remote-throughput\" transaction=\"xa\"/>" | sed -e ':a;N;$!ba;s|\n|\\n|g'
+generate_remote_artemis_connection_factory() {
+    declare name="${1}"
+    declare username="${2}"
+    declare password="${3}"
+    declare connectors="${4}"
+    declare entries="${5}"
+    echo "<pooled-connection-factory name=\"${name}\" user=\"${username}\" password=\"${password}\" entries=\"${entries}\" connectors=\"${connectors}\" transaction=\"xa\"/>" | sed -e ':a;N;$!ba;s|\n|\\n|g'
 }
 
-# $1 object type - queue / topic
-# $2 object name - MyQueue / MyTopic
 # <!-- ##AMQ7_CONFIG_PROPERTIES## -->
-function generate_remote_artemis_property() {
-    echo "<property name=\"${1}.${2}\" value=\"${2}\"/>" | sed -e ':a;N;$!ba;s|\n|\\n|g'
+generate_remote_artemis_property() {
+    declare object_type="${1}"
+    declare object_name="${2}"
+    echo "<property name=\"${object_type}.${object_name}\" value=\"${object_name}\"/>" | sed -e ':a;N;$!ba;s|\n|\\n|g'
 }
 
-# $1 - remote context - remoteContext
-# $2 - object name - MyQueue / MyTopic etc
 # <!-- ##AMQ_LOOKUP_OBJECTS## -->
-function generate_remote_artemis_lookup() {
-    echo "<lookup name=\"java:/${2}\" lookup=\"java:global/${1}/${2}\"/>" | sed -e ':a;N;$!ba;s|\n|\\n|g'
+generate_remote_artemis_lookup() {
+    declare remote_context="${1}"
+    declare object_name="${2}"
+    echo "<lookup name=\"java:/${object_name}\" lookup=\"java:global/${remote_context}/${object_name}\"/>" | sed -e ':a;N;$!ba;s|\n|\\n|g'
 }
 
-# $1 - name - messaging-remote-throughput
-# $2 - remote hostname
-# $3 - remote port
 # <!-- ##AMQ_MESSAGING_SOCKET_BINDING## -->
-function generate_remote_artemis_socket_binding() {
-    echo "<outbound-socket-binding name=\"${1}\">
-            <remote-destination host=\"${2}\" port=\"${3}\"/>
+generate_remote_artemis_socket_binding() {
+    declare name="${1}"
+    declare remote_host="${2}"
+    declare remote_port="${3}"
+    echo "<outbound-socket-binding name=\"${name}\">
+            <remote-destination host=\"${remote_host}\" port=\"${port}\"/>
          </outbound-socket-binding>" | sed -e ':a;N;$!ba;s|\n|\\n|g'
 }
 
-# Arguments:
-# $1 - physical name
-# $2 - jndi name
-# $3 - class
```

Comment:
Nit: if these were put back before current L234 the diff would be simpler

Reply:
Never mind; i see you are replacing comments with 'declare' which makes perfect sense since the comment would be redundant